media-libs/opencv/opencv-4.9.0.ebuild

```diff
@@ -341,6 +341,10 @@ src_prepare() {
 		cd "${WORKDIR}/${PN}_contrib-${PV}" || die
 		eapply "${FILESDIR}/${PN}_contrib-4.8.1-rgbd.patch"
 		eapply "${FILESDIR}/${PN}_contrib-4.8.1-NVIDIAOpticalFlowSDK-2.0.tar.gz.patch"
+		if has_version ">=dev-util/nvidia-cuda-toolkit-12.4" && use cuda; then
+			# TODO https://github.com/NVIDIA/cccl/pull/1522
+			eapply "${FILESDIR}/${PN}_contrib-4.9.0-cuda-12.4.patch"
+		fi
```

Comment:
What is the point of this if we restrict the version to lower then 12.4?

Reply:
None yet. It's just the start of the work to support CUDA-12.4. Still needs the fixes for the cccl mishap. But I feel like it's easier to start from there for anyone looking  then to have to try to solve that as well.